app/controllers/auth/PertaxAuthAction.scala

```diff
@@ -66,10 +67,14 @@ class PertaxAuthActionImpl @Inject() (
         pertaxConnector
           .pertaxPostAuthorise()
           .fold(
-            { _: UpstreamErrorResponse =>
-              Future.successful(
-                Some(InternalServerError(internalServerErrorView(appConfig)))
-              )
+            { error: UpstreamErrorResponse =>
+              if (error.statusCode == 401) {
```

Comment:
For clarity can you use the constant UNAUTHORIZED? Also, could slightly shorten and clarify code like this:
          .fold(
            error =>
              Future.successful(Some(
                error.statusCode match {
                  case 401 => Redirect(routes.UnauthorisedController.loginGG())
                  case _ => InternalServerError(internalServerErrorView(appConfig))
                }
              )),